test/preview/components/builders/test_dynamic_prompt_builder.py

```diff
@@ -0,0 +1,139 @@
+    )
+    def test_initialization(self, expected_runtime_variables: List[str], chat_mode: bool):
+        builder = DynamicPromptBuilder(expected_runtime_variables, chat_mode)
+        assert builder.expected_runtime_variables == expected_runtime_variables
+        assert builder.chat_mode == chat_mode
+
+        # regardless of the chat mode
+        # we have inputs that contain: prompt_source, template_variables + expected_runtime_variables
+        expected_keys = set(expected_runtime_variables + ["prompt_source", "template_variables"])
+        assert set(builder.__canals_input__.keys()) == expected_keys
+
+        # response is always prompt regardless of chat mode
+        assert set(builder.__canals_output__.keys()) == {"prompt"}
+
+        # prompt_source is a list of ChatMessage or a string
+        assert builder.__canals_input__["prompt_source"].type == Union[List[ChatMessage], str]
+
+        # output is always prompt, but the type is different depending on the chat mode
+        if chat_mode:
+            assert builder.__canals_output__["prompt"].type == List[ChatMessage]
+        else:
+            assert builder.__canals_output__["prompt"].type == str
+
+    def test_to_dict_method_returns_expected_dictionary(self):
+        expected_runtime_variables = ["var1", "var2", "var3"]
+        chat_mode = True
+        builder = DynamicPromptBuilder(expected_runtime_variables, chat_mode)
+        expected_dict = {
+            "type": "haystack.preview.components.builders.dynamic_prompt_builder.DynamicPromptBuilder",
+            "init_parameters": {"expected_runtime_variables": expected_runtime_variables, "chat_mode": chat_mode},
+        }
+        assert builder.to_dict() == expected_dict
+
+    def test_processing_a_simple_template_with_provided_variables(self):
+        expected_runtime_variables = ["var1", "var2", "var3"]
+        chat_mode = True
+
+        builder = DynamicPromptBuilder(expected_runtime_variables, chat_mode)
+
+        template = "Hello, {{ name }}!"
+        template_variables = {"name": "John"}
+        expected_result = "Hello, John!"
+
+        assert builder._process_simple_template(template, template_variables) == expected_result
+
+    def test_processing_a_simple_template_with_invalid_template(self):
+        expected_runtime_variables = ["var1", "var2", "var3"]
+        chat_mode = True
+
+        builder = DynamicPromptBuilder(expected_runtime_variables, chat_mode)
+
+        template = "Hello, {{ name }!"
+        template_variables = {"name": "John"}
+        with pytest.raises(TemplateSyntaxError):
+            builder._process_simple_template(template, template_variables)
+
+    def test_processing_a_simple_template_with_missing_variables(self):
+        expected_runtime_variables = ["var1", "var2", "var3"]
+
+        builder = DynamicPromptBuilder(expected_runtime_variables, False)
+
+        with pytest.raises(ValueError, match="requires specific template variables that are missing"):
```

Comment:
Let's not match error messages, it's more annoying than useful most of the times.

```suggestion
        with pytest.raises(ValueError):
```